test/LibraryManager.IntegrationTest/AddCientSideLibrariesFromUITests.cs

```diff
@@ -0,0 +1,115 @@
+            }
+        }
+
+        [TestMethod]
+        public void InstallClientSideLibraries_FromProjectRoot_SmokeTest()
+        {
+            SetLibraryAndClickInstall(_projectName, "jquery-validate@1.17.0");
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "lib", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            string manifestContents = @"{
+  ""version"": ""1.0"",
+  ""defaultProvider"": ""cdnjs"",
+  ""libraries"": [
+    {
+      ""library"": ""jquery-validate@1.17.0"",
+      ""destination"": ""wwwroot/lib/jquery-validate/""
+    }
+  ]
+}";
+            Helpers.FileIO.WaitForRestoredFiles(pathToLibrary, expectedFiles, caseInsensitive: true, timeout: 20000);
+            Assert.AreEqual(manifestContents, File.ReadAllText(_pathToLibmanFile));
+        }
+
+        [TestMethod]
+        public void InstallClientSideLibraries_FromFolder_SmokeTest()
+        {
+            SetLibraryAndClickInstall("wwwroot", "jquery-validate@1.17.0");
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            string manifestContents = @"{
+  ""version"": ""1.0"",
+  ""defaultProvider"": ""cdnjs"",
+  ""libraries"": [
+    {
+      ""library"": ""jquery-validate@1.17.0"",
+      ""destination"": ""wwwroot/jquery-validate/""
+    }
+  ]
+}";
+            Helpers.FileIO.WaitForRestoredFiles(pathToLibrary, expectedFiles, caseInsensitive: true, timeout: 20000);
+            Assert.AreEqual(manifestContents, File.ReadAllText(_pathToLibmanFile));
+        }
+
+        private void SetLibraryAndClickInstall(string nodeName, string library)
+        {
+            SolutionExplorer.Select(nodeName);
+            InstallDialogTestService installDialogTestService = VisualStudio.Get<InstallDialogTestService>();
+            InstallDialogTestExtension installDialogTestExtenstion = installDialogTestService.OpenDialog();
+
+            installDialogTestExtenstion.SetLibrary(library);
+            installDialogTestExtenstion.WaitForFileSelections();
+            installDialogTestExtenstion.ClickInstall();
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            _libManConfig = _webProject[_libman];
+
+            if (_libManConfig != null)
+            {
+                _libManConfig.Open();
+
+                Editor.Selection.SelectAll();
+                Editor.KeyboardCommands.Backspace();
```

Comment:
>Editor.KeyboardCommands.Backspace(); [](start = 16, length = 36)

nit: necessary?

Reply:
This was just doing a delete. We need to SelectAll, delete and insert. Basically select and replace. I don't see an api that would directly support this. Changing it to below:

```
Editor.Selection.SelectAll();
Editor.KeyboardCommands.Delete();
Editor.Edit.InsertTextInBuffer(_initialLibmanFileContent);
```

---
In reply to: [205460210](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205460210) [](ancestors = 205460210)